transport-native-kqueue/src/main/c/netty_kqueue_bsdsocket.c

```diff
@@ -83,6 +83,57 @@ static jlong netty_kqueue_bsdsocket_sendFile(JNIEnv* env, jclass clazz, jint soc
     return res < 0 ? -err : 0;
 }
 
+static jint netty_kqueue_bsdsocket_connectx(JNIEnv* env,
+        jint socketFd,
+        jint socketInterface,
+        jboolean sourceIPv6, jbyteArray sourceAddress, jint sourceScopeId, jint sourcePort,
+        jboolean destinationIPv6, jbyteArray destinationAddress, jint destinationScopeId, jint destinationPort,
+        jint flags,
+        jlong iovAddress, jint iovCount, jint iovDataLength) {
+#ifdef __APPLE__ // connectx(2) is only defined on Darwin.
+    sa_endpoints_t endpoints;
+    endpoints.sae_srcif = (unsigned int) socketInterface;
+    endpoints.sae_srcaddr = NULL;
+    endpoints.sae_srcaddrlen = 0;
+    endpoints.sae_dstaddr = NULL;
+    endpoints.sae_dstaddrlen = 0;
+
+    struct sockaddr_storage srcaddr;
+    socklen_t srcaddrlen;
+    struct sockaddr_storage dstaddr;
+    socklen_t dstaddrlen;
+
+    if (NULL != sourceAddress) {
+        if (-1 == netty_unix_socket_initSockaddr(env,
+                sourceIPv6, sourceAddress, sourceScopeId, sourcePort, &srcaddr, &srcaddrlen)) {
+            return -1;
+        }
+        endpoints.sae_srcaddr = (const struct sockaddr*) &srcaddr;
+        endpoints.sae_srcaddrlen = srcaddrlen;
+    }
+    if (NULL != destinationAddress) {
+        if (-1 == netty_unix_socket_initSockaddr(env,
+                destinationIPv6, destinationAddress, destinationScopeId, destinationPort, &dstaddr, &dstaddrlen)) {
+            return -1;
+        }
+        endpoints.sae_dstaddr = (const struct sockaddr*) &dstaddr;
+        endpoints.sae_dstaddrlen = dstaddrlen;
+    }
+
+    int socket = (int) socketFd;
```

Comment:
nit: these calls can be inlined, may make it easier for compiler to remove intermediate storage and less intermediate variable noise in the code if they are only used once.

Reply:
I don't understand this concern with local variables.

Reply:
fine to leave as is but the goal would be to decrease noise in the code and aid readability (why are there multiple variables, can their values change independently, what is the memory lifetime/location of each, etc.).